common/djangoapps/third_party_auth/management/commands/saml.py

```diff
@@ -107,52 +111,75 @@ def _check_provider_configurations(self):
-                        self.stdout.write(
-                            f"[WARNING] {provider_info} "
-                            f"has outdated SAML config (id={provider_config.saml_configuration_id} which "
-                            f"should be updated to the current SAML config (id={current_config.id})."
-                        )
-                        outdated_count += 1
+                if current_config and (current_config.id != provider_config.saml_configuration_id):
+                    # Resolution: Update the provider's saml_configuration_id to the current config ID
+                    self.stdout.write(
+                        f"[WARNING] {provider_info} "
+                        f"has outdated SAML config (id={provider_config.saml_configuration_id}) which "
+                        f"should be updated to the current SAML config (id={current_config.id})."
+                    )
+                    outdated_count += 1
 
+                # Check site ID match
                 if provider_config.saml_configuration.site_id != provider_config.site_id:
                     config_site_id = provider_config.saml_configuration.site_id
-                    provider_site_id = provider_config.site_id
+                    # Resolution: Create a new SAML configuration for the correct site
+                    # or move the provider to the matching site
                     self.stdout.write(
                         f"[WARNING] {provider_info} "
-                        f"SAML config (id={provider_config.saml_configuration_id}, site_id={config_site_id}) "
-                        "does not match the provider's site_id."
+                        f"SAML config (id={provider_config.saml_configuration_id}, "
+                        f"site_id={config_site_id}) does not match the provider's site_id."
                     )
                     site_mismatch_count += 1
 
-                saml_configuration_slug = provider_config.saml_configuration.slug
-                provider_config_slug = provider_config.slug
-
-                if saml_configuration_slug not in (provider_config_slug, 'default'):
+                # Check slug match
+                if provider_config.saml_configuration.slug not in (provider_config.slug, 'default'):
+                    config_id = provider_config.saml_configuration_id
+                    saml_configuration_slug = provider_config.saml_configuration.slug
+                    # Resolution: This is informational only - provider can use
+                    # a different slug configuration
                     self.stdout.write(
-                        f"[WARNING] {provider_info} "
-                        f"SAML config (id={provider_config.saml_configuration_id}, slug='{saml_configuration_slug}') "
+                        f"[INFO] {provider_info} "
+                        f"SAML config (id={config_id}, slug='{saml_configuration_slug}') "
                         "does not match the provider's slug."
```

Comment:
1. Edit to make more readable.
```suggestion
                        f"[INFO] {provider_info} has "
                        f"SAML config (id={config_id}, slug='{saml_configuration_slug}') "
                        "that does not match the provider's slug."
```
2. Add `DISABLED ` before SAML config if it is disabled.

Reply:
resolved the info messages